pkg/cluster/util.go

```diff
@@ -507,38 +507,38 @@ func (c *Cluster) roleLabelsSet(shouldAddExtraLabels bool, role PostgresRole) la
 
 func (c *Cluster) dnsName(role PostgresRole) string {
 	var dnsString string
+
 	if role == Master {
 		dnsString = c.masterDNSName()
 	} else {
 		dnsString = c.replicaDNSName()
 	}
 
-	// when cluster name starts with teamId prefix create an extra DNS entry
-	// to support the old format when prefix contraint was enabled (but is disabled now)
-	if !c.OpConfig.EnableTeamIdClusternamePrefix {
-		clusterNameWithoutTeamPrefix, _ := acidv1.ExtractClusterName(c.Name, c.Spec.TeamID)
-		if clusterNameWithoutTeamPrefix != "" {
-			if role == Replica {
-				clusterNameWithoutTeamPrefix = fmt.Sprintf("%s-repl", clusterNameWithoutTeamPrefix)
-			}
-			dnsString = fmt.Sprintf("%s,%s", dnsString, c.oldDNSFormat(clusterNameWithoutTeamPrefix))
+	// if cluster name starts with teamID we might need to provide backwards compatibility
+	clusterNameWithoutTeamPrefix, _ := acidv1.ExtractClusterName(c.Name, c.Spec.TeamID)
+	if clusterNameWithoutTeamPrefix != "" {
+		if role == Replica {
+			clusterNameWithoutTeamPrefix = fmt.Sprintf("%s-repl", clusterNameWithoutTeamPrefix)
 		}
+		dnsString = fmt.Sprintf("%s,%s", dnsString, c.oldDNSFormat(clusterNameWithoutTeamPrefix))
 	}
 
 	return dnsString
 }
 
 func (c *Cluster) masterDNSName() string {
 	return strings.ToLower(c.OpConfig.MasterDNSNameFormat.Format(
-		"cluster", c.Spec.ClusterName,
+		"cluster", c.Name,
 		"namespace", c.Namespace,
+		"team", c.teamName(),
```

Comment:
If somebody still has the old {clustername}.{team}.{hostedzone} template in the config, we'll make sure it continues to work.